File: app/utils/DatabaseUtil.java
package utils;

import com.mongodb.DB;
import com.mongodb.MongoClient;
import net.vz.mongodb.jackson.JacksonDBCollection;

import java.net.UnknownHostException;

public class DatabaseUtil {

    private static MongoClient mongoClient;

    public static DB getDB() {

        try {
            mongoClient = new MongoClient( );

            return mongoClient.getDB("database");

        } catch (UnknownHostException e) {
            return null;
        }
    }

    public static JacksonDBCollection getCollection(String collection, Class clazz) {

        try {

            return JacksonDBCollection.wrap(getDB().getCollection(collection), clazz);

        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }

    }

    public static Object getEntityById(String collection, Class clazz, String id) {

        try {
            JacksonDBCollection jacksonDBCollection = JacksonDBCollection.wrap(getDB().getCollection(collection), clazz);
            return jacksonDBCollection.findOneById(id);
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }

    }
}

File: app/utils/DatabaseUtil.scala
package utils

import reactivemongo.api._
import scala.concurrent.ExecutionContext.Implicits.global

object DatabaseUtil {

  def connect(): reactivemongo.api.MongoConnection = {
    // gets an instance of the driver
    // (creates an actor system)
    val driver = new MongoDriver
    driver.connection(List("localhost"))
  }

  def getCollection(collectionName: String) = {

    // Gets a reference to the database "plugin"
    val connection = connect()

    // Gets a reference to the database "plugin"
    val db = connection("scheinwelt")

    // Gets a reference to the collection "acoll"
    // By default, you get a BSONCollection.
    val collection = db(collectionName)

  }

  //    def getEntityById[T](collectionName: String, id: String):Any = {
  //
  //        try {
  //            val coll = db(collectionName)
  //            coll.findOneById(id)
  //        } catch {
  //          case e: Exception => e.printStackTrace()
  //        }
  //
  //    }
}

